fhir-model/src/main/java/com/ibm/fhir/model/path/FHIRPathQuantityValue.java

```diff
@@ -132,24 +132,25 @@ public FHIRPathQuantityValue subtract(FHIRPathQuantityValue quantityValue) {
 
     @Override
     public boolean isComparableTo(FHIRPathNode other) {
-        if (other instanceof FHIRPathQuantityValue) {
-            return unit.equals(((FHIRPathQuantityValue) other).unit());
+        if (other instanceof FHIRPathQuantityValue || other.getValue() instanceof FHIRPathQuantityValue) {
+            FHIRPathQuantityValue quantityValue = (other instanceof FHIRPathQuantityValue) ? (FHIRPathQuantityValue) other : (FHIRPathQuantityValue) other.getValue();
+            return unit.equals(quantityValue.unit());
         }
-        if (other.getValue() instanceof FHIRPathQuantityValue) {
-            return unit.equals(((FHIRPathQuantityValue) other.getValue()).unit());
-        }
-        return (other instanceof FHIRPathNumberValue) || (other.getValue() instanceof FHIRPathNumberValue);
+        return (other instanceof FHIRPathNumberValue) || 
+                (other.getValue() instanceof FHIRPathNumberValue);
     }
 
     @Override
     public int compareTo(FHIRPathNode other) {
         if (!isComparableTo(other)) {
             throw new IllegalArgumentException();
         }
-        if (other instanceof FHIRPathQuantityValue) {
-            return value.compareTo(((FHIRPathQuantityValue) other).value());
+        if (other instanceof FHIRPathQuantityValue || other.getValue() instanceof FHIRPathQuantityValue) {
+            FHIRPathQuantityValue quantityValue = (other instanceof FHIRPathQuantityValue) ? (FHIRPathQuantityValue) other : (FHIRPathQuantityValue) other.getValue();
+            return value.compareTo(quantityValue.value());
         }
-        return value.compareTo(((FHIRPathQuantityValue) other.getValue()).value());
+        FHIRPathNumberValue numberValue = (other instanceof FHIRPathNumberValue) ? (FHIRPathNumberValue) other : (FHIRPathNumberValue) other.getValue();
+        return value.compareTo(numberValue.decimal());
```

Comment:
hum...  there seems to be similar conversions and checks in the blocks

Reply:
compareTo is protected by isComparableTo (even though isComparableTo is always called before compareTo in the FHIRPath evaluator)